parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -196,10 +196,25 @@ configureCNIIPTables() {
     fi
 }
 
+ensureContainerRuntime() {
+    {{if IsDockerContainerRuntime}}
+        ensureDocker
+    {{end}}
+    {{if NeedsContainerd}}
+        ensureContainerd
+    {{end}}
+}
+
 {{if NeedsContainerd}}
 ensureContainerd() {
-    echo "Starting cri-containerd service..."
-    systemctlEnableAndStart containerd || exit $ERR_SYSTEMCTL_START_FAIL
+  wait_for_file 1200 1 /etc/systemd/system/containerd.service.d/exec_start.conf || exit $ERR_FILE_WATCH_TIMEOUT
+  wait_for_file 1200 1 /etc/containerd/config.toml || exit $ERR_FILE_WATCH_TIMEOUT
+  {{if IsKubenet }}
+  wait_for_file 1200 1 /etc/sysctl.d/11-containerd.conf || exit $ERR_FILE_WATCH_TIMEOUT
+  retrycmd_if_failure 120 5 25 sysctl --system || exit $ERR_SYSCTL_RELOAD
+  {{end}}
+  systemctl is-active --quiet docker && (systemctl_stop 20 30 120 docker || exit $ERR_SYSTEMD_DOCKER_STOP_FAIL)
```

Comment:
any thoughts on systemctl stop vs disable here?

Reply:
yea good suggestion - that seems better